tssc/step_implementers/shared/openscap_generic.py

```diff
@@ -0,0 +1,656 @@
+            ) from error
+
+        return mount_path
+
+    @staticmethod
+    def __get_oscap_document_type(oscap_input_file):
+        """Gets the OpenSCAP document type for a given input file.
+
+        Parameters
+        ----------
+        oscap_input_file : path
+            Path to OSCAP file to determine the OpenSCAP document type of.
+
+        Returns
+        -------
+        str
+            OpenSCAP document type. For example:
+            * Source Data Stream
+            * XCCDF Checklist
+            * OVAL Definitions
+
+        Raises
+        ------
+        ErrorReturnCode
+            If unexpected error occurred
+        """
+
+        oscap_document_type = None
+        try:
+            oscap_info_out_buff = StringIO()
+            sh.oscap.info( # pylint: disable=no-member
+                oscap_input_file,
+                _out=oscap_info_out_buff
+            )
+            oscap_info_out = oscap_info_out_buff.getvalue().rstrip()
+            oscap_document_type_match = OpenSCAPGeneric.OSCAP_INFO_DOC_TYPE_PATTERN.search(
+                oscap_info_out
+            )
+            oscap_document_type = oscap_document_type_match.groupdict()['doctype']
+        except sh.ErrorReturnCode as error:
+            raise RuntimeError(
+                f"Unexpected error getting document type of oscap input file"
+                f" ({oscap_input_file}): {error}"
+            ) from error
+
+        return oscap_document_type
+
+    @staticmethod
+    def __get_oscap_eval_type_based_on_document_type(oscap_document_type):
+        """Given an OSCAP document type returns the type of oscap eval that should be used.
+
+        Parameters
+        ----------
+        oscap_document_type : str
+            OSCAP Document type to get the oscap eval type for.
+
+        Returns
+        -------
+        str
+            OSCAP eval type to perform on document with given oscap document type.
+        """
+        oscap_eval_type = None
+
+        if oscap_document_type == 'Source Data Stream':
+            oscap_eval_type = 'xccdf'
+        elif oscap_document_type == 'XCCDF Checklist':
+            oscap_eval_type = 'xccdf'
+        elif oscap_document_type == 'OVAL Definitions':
+            oscap_eval_type = 'oval'
+
+        return oscap_eval_type
+
+    @staticmethod
+    def __run_oscap_scan( # pylint: disable=too-many-arguments,too-many-locals,too-many-branches,too-many-statements
+        buildah_unshare_comand,
+        oscap_eval_type,
+        oscap_input_file,
+        oscap_out_file_path,
+        oscap_xml_results_file_path,
+        oscap_html_report_path,
+        container_mount_path,
+        oscap_profile=None,
+        oscap_tailoring_file=None,
+        oscap_fetch_remote_resources=True
+    ):
+        """Run an oscap scan in the context of a buildah unshare to run "rootless".
+
+        Parameters
+        ----------
+        buildah_unshare_comand : sh.buildah.unsahre.bake()
```

Comment:
Nit:
`buildah_unshare_comand : sh.buildah.unsahre.bake()`

to

`buildah_unshare_comand : sh.buildah.unshare.bake()`